Tasks/KubernetesV1/src/clusterconnection.ts

```diff
@@ -64,14 +66,24 @@ export default class ClusterConnection {
             kubeconfig = await this.getKubeConfig(connectionType);
         }
 
-        return this.initialize().then(() => {
+        return this.initialize().then(async () => {
             if (kubeconfig)
             {
                 this.kubeconfigFile = path.join(this.userDir, "config");
                 fs.writeFileSync(this.kubeconfigFile, kubeconfig);
             }
 
             process.env["KUBECONFIG"] = this.kubeconfigFile;
+
+            try {
+              const kubelogin = new Kubelogin(this.userDir);
+              if (kubelogin.isAvailable()) {
+                tl.debug('Kubelogin is installed. Converting kubeconfig.');
+                await kubelogin.login(tl.getInput('azureSubscriptionEndpoint', false));
+              }
+            } catch (err) {
```

Comment:
>catch [](http://example.com/codeflow?start=14&length=5)

it's better to avoid try{} catch{} if there is some other way to handle the login failure.  

if try{} catch must be used, please try to narrow it to just the statement that will throw and leave other statements out to avoid masking true bugs

Reply:
I only moved login() under try{] catch {}. Thanks!